lib/std/Io/Threaded.zig

```diff
@@ -420,23 +421,24 @@ const AsyncClosure = struct {
 
     fn resultPointer(ac: *AsyncClosure) [*]u8 {
         const base: [*]u8 = @ptrCast(ac);
-        return base + ac.result_offset;
+        const closure_size_with_padding = @sizeOf(AsyncClosure) + ac.context_alignment.forward(@alignOf(AsyncClosure));
+        return @ptrFromInt(ac.result_alignment.forward(@intFromPtr(base + closure_size_with_padding)));
     }
```

Comment:
This looks wrong to me, the code is not aware of the context length. `resultPointer()` will return a pointer into the middle of the context. `AsyncClosure` needs to remember either the computed result offset, or the context alignment + context length + result alignment to recompute it.